src/main/java/net/dv8tion/jda/api/utils/messages/MessageEditBuilder.java

```diff
@@ -0,0 +1,322 @@
+     *
+     * @return A new MessageEditBuilder instance with the applied data
+     *
+     * @see    #applyMessage(Message)
+     */
+    @Nonnull
+    public static MessageEditBuilder fromMessage(@Nonnull Message message)
+    {
+        return new MessageEditBuilder().applyMessage(message);
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder mentionRepliedUser(boolean mention)
+    {
+        super.mentionRepliedUser(mention);
+        set |= MENTIONS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder allowedMentions(@Nullable Collection<Message.MentionType> allowedMentions)
+    {
+        super.allowedMentions(allowedMentions);
+        set |= MENTIONS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder mention(@Nonnull IMentionable... mentions)
+    {
+        super.mention(mentions);
+        set |= MENTIONS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder mentionUsers(@Nonnull String... userIds)
+    {
+        super.mentionUsers(userIds);
+        set |= MENTIONS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder mentionRoles(@Nonnull String... roleIds)
+    {
+        super.mentionRoles(roleIds);
+        set |= MENTIONS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setAttachments(@Nullable Collection<? extends AttachedFile> attachments)
+    {
+        this.attachments.clear();
+        set |= ATTACHMENTS;
+        if (attachments != null)
+            this.attachments.addAll(attachments);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends AttachedFile> getAttachments()
+    {
+        return Collections.unmodifiableList(attachments);
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder replace(boolean isReplace)
+    {
+        this.replace = isReplace;
+        return this;
+    }
+
+    @Override
+    public boolean isReplace()
+    {
+        return replace;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setContent(@Nullable String content)
+    {
+        super.setContent(content);
+        set |= CONTENT;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setEmbeds(@Nonnull Collection<? extends MessageEmbed> embeds)
+    {
+        super.setEmbeds(embeds);
+        set |= EMBEDS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder setComponents(@Nonnull Collection<? extends LayoutComponent> components)
+    {
+        super.setComponents(components);
+        set |= COMPONENTS;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditBuilder applyData(@Nonnull MessageEditData data)
+    {
+        Checks.notNull(data, "Data");
+        this.set = data.getFlags();
```

Comment:
I understand what this is mutating the flags so that `isSet` resolves correctly, but honestly this is confusing.
I wish it was doing `data.isSet(CONTENT)` instead of `this.isSet(CONTENT)` because it semantically doesn't follow.

We can package protected expose `isSet` on `MessageEditData` to solve this problem.